com.unity.multiplayer.mlapi/Runtime/Messaging/MessageQueue/MessageQueueProcessor.cs

```diff
@@ -314,8 +317,17 @@ private void SendCallback(ulong clientId, MessageBatcher.SendStream sendStream)
             var length = (int)sendStream.Buffer.Length;
             var bytes = sendStream.Buffer.GetBuffer();
             var sendBuffer = new ArraySegment<byte>(bytes, 0, length);
-
-            m_MessageQueueContainer.NetworkManager.NetworkConfig.NetworkTransport.Send(clientId, sendBuffer, sendStream.NetworkChannel);
+            
+            var channel = sendStream.NetworkChannel;
+            // If the length is greater than the fragmented threshold, switch to a fragmented channel.
+            // This is kind of a hack to get around issues with certain usages patterns on fragmentation with UNet.
+            // We send everything unfragmented to avoid those issues, and only switch to the fragmented channel
+            // if we have no other choice.
+            if (length > k_FragmentationThreshold)
+            {
+                channel = NetworkChannel.Fragmented;
```

Comment:
I don't think you can do that. As far as I'm aware there's no ordering guarantees **between** channels. As such, stuff that was already sent on the non-fragmented channel could arrive after later fragmented message on the fragmented channel. If we get there and have this issue, I'd say let's switch it all to fragmented, and drop the condition.

Reply:
Unfortunately, the message that hits this is the very first message we send, so... that'd just result in everything being fragmented.

Reply:
So what does this change do then? We still end up sending everything on the reliableFragementedSequenced channel which is what was happening already before?

Reply:
ConnectionAccepted gets switched to Fragmented. I haven't seen any other messages that get too large for ReliableSequenced so far, though it's possible there are some. But this changelist basically changes everything to ReliableSequenced... and then this if check here is just to handle ConnectionAccepted and nothing else.

In my serialization RFC I'm going to rework that and any other messages that can get that large to be multiple messages instead of one mega-message, so this should stop being an issue for internal messages and will only matter for user RPCs or CustomMessages that get too large.

Reply:
So my response here is to say:

- This is a short-term fix to address the issues we're seeing in develop now, and will be torn out again soon to be fixed a better way with the serialization refactor
- For internal messages, we won't worry about ordering because we'll make it a requirement going forward to keep them within MTU size and not use fragmented messages
- For user-generated messages (RPCs) we actually currently don't accept anything larger than MTU, but in the serialization refactor we'll start accepting them. We'll make sure it's documented that large messages that need to be fragmented will be sequenced separately from small ones that don't need to be.

Reply:
I don't see why it's not possible to simply use `channel = NetworkChannel.Fragmented` always. What's the reason against ?

Reply:
See my comment above in reply to Luke. Fragmented is not able to handle high volumes. It starts dropping messages.

Reply:
yikes. Please, please, let's not forget this. But yeah, as a temporary band-aid...